Neos.Neos/Classes/Domain/Model/Site.php

```diff
@@ -372,7 +380,16 @@ public function emitSiteChanged()
 
     public function getConfiguration(): SiteConfiguration
     {
-        // we DO NOT want recursive merge here
-        return SiteConfiguration::fromArray($this->sitesConfiguration[$this->nodeName] ?? $this->sitesConfiguration['*']);
+        $siteSettingsPath = array_key_exists($this->nodeName, $this->sitesConfiguration) ? $this->nodeName : '*';
+        $siteSettings = $this->sitesConfiguration[$siteSettingsPath];
+        if (isset($siteSettings['preset'])) {
+            is_string($siteSettings['preset']) || throw new \RuntimeException(sprintf('Invalid "preset" configuration for "Neos.Neos.sites.%s". Expected string, got: %s', $siteSettingsPath, get_debug_type($siteSettings['preset'])), 1699785648);
```

Comment:
> As far as i know this is against our PSR 12 standard

No, that's perfectly valid PSR 12.
Feel free to change it, but IMO it's really a non-issue

Reply:
Yeah no problem no need to split hairs, i overheard that christian absolutely hates this ^^

Reply:
Then he should speak up :)
But it's in line with PSR 12 AFAIK and I find this:
```php
if (isset($siteSettings['preset'])) {
    if (!is_string($siteSettings['preset'])) {
        throw new \RuntimeException(sprintf('Invalid "preset" configuration for "Neos.Neos.sites.%s". Expected string, got: %s', $siteSettingsPath, get_debug_type($siteSettings['preset'])), 1699785648);
    }
    if (!isset($this->sitePresetsConfiguration[$siteSettings['preset']]) || !is_array($this->sitePresetsConfiguration[$siteSettings['preset']])) {
        throw new \RuntimeException(sprintf('Site settings "Neos.Neos.sites.%s" refer to a preset "%s", but no corresponding preset is configured', $siteSettingsPath, $siteSettings['preset']), 1699785736);
    }
}
```
harder to comprehend than:
```php
if (isset($siteSettings['preset'])) {
    is_string($siteSettings['preset']) || throw new \RuntimeException(sprintf('Invalid "preset" configuration for "Neos.Neos.sites.%s". Expected string, got: %s', $siteSettingsPath, get_debug_type($siteSettings['preset'])), 1699785648);
    !is_array($this->sitePresetsConfiguration[$siteSettings['preset']] ?? null) || throw new \RuntimeException(sprintf('Site settings "Neos.Neos.sites.%s" refer to a preset "%s", but no corresponding preset is configured', $siteSettingsPath, $siteSettings['preset']), 1699785736);
}
```

However. I just realized, that we currently mix both versions :)
I'll change both to the first style because, frankly, that's just more common